tests/kv_list_group_keys_eqc.erl

```diff
@@ -0,0 +1,198 @@
+-module(kv_list_group_keys_eqc).
+
+-ifdef(EQC).
+
+-compile(export_all).
+-include_lib("eqc/include/eqc.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-behaviour(riak_test).
+-export([confirm/0]).
+
+-define(CONFIG,
+        [{riak_core, [{handoff_concurrency, 10},
+                      {vnode_management_timer, 1000}]},
+         {riak_kv, [{storage_backend, riak_kv_eleveldb_backend}]}]).
+-define(CLUSTER_SIZE, 5).
+-define(PHOTOS_PREFIX, <<"photos/2017/">>).
+-define(MONTHS, [<<"January">>, <<"February">>, <<"March">>, <<"April">>,
+                 <<"May">>, <<"June">>, <<"July">>, <<"August">>,
+                 <<"September">>, <<"October">>, <<"November">>, <<"December">>]).
+
+-record(group_keys_result, {
+          keys = [] :: list(),
+          common_prefixes = [] :: list()
+         }).
+
+confirm() ->
+    Cluster = setup_cluster(?CLUSTER_SIZE),
+    put_objects(Cluster),
+    quickcheck(?MODULE:prop_group_keys_basic(Cluster)),
+    quickcheck(?MODULE:prop_group_keys_prefix(Cluster)),
+    quickcheck(?MODULE:prop_group_keys_delimiter(Cluster)),
+    quickcheck(?MODULE:prop_group_keys_prefix_delimiter(Cluster)),
+    rt:clean_cluster(Cluster),
+    pass.
+
+%% ====================================================================
+%% EQC generators
+%% ====================================================================
+gen_max_keys() ->
+    NumKeys = length(all_keys()),
+    choose(erlang:max(1, NumKeys div 10), NumKeys).
+
+%% ====================================================================
+%% EQC Properties
+%% ====================================================================
+
+quickcheck(Property) ->
+    ?assert(eqc:quickcheck(eqc:numtests(500, Property))).
+
+forall_buckets_and_max_keys(Property) ->
+    ?FORALL(Bucket, oneof(buckets()),
```

Comment:
Not critiquing, but just curious, since I'm not quite an expert-level EQC user myself: what's the purpose of using the nested `?FORALL` statements instead of just doing something like:
```
?FORALL({Bucket, MaxKeys}, {oneof(buckets()), gen_max_keys()},
    collect(MaxKeys, Property(Bucket, MaxKeys))).
```